### PR TITLE
feat: epoch 4 LM incentives

### PR DESCRIPTION
**Pull request by lilPlumberBoy**
Adding scripts for both ops and community multsig transactions for epoch 4 incentives: https://docs.cove.finance/ecosystem/token/liquidity-mining#epoch-4

### CODE REVIEW

File: script/ops/Ops_2024_08_22.s.sol [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+contract Script is OpsMultisigScript {
+    function run(bool shouldSend) public override {
+        super.run(shouldSend);
+
+        address coveToken = deployer.getAddress("CoveToken");
+        address coveYfiRewardsGauge = deployer.getAddress("CoveYfiRewardsGauge");
+        address coveYfiRewardsGaugeRewardForwarder = deployer.getAddress("CoveYFIRewardsGaugeRewardForwarder");
+        CoveYearnGaugeFactory factory = CoveYearnGaugeFactory(deployer.getAddress("CoveYearnGaugeFactory"));
+        CoveYearnGaugeFactory.GaugeInfo[] memory info = factory.getAllGaugeInfo(9, 0);
```
Comment (penandlim) [inline script/ops/Ops_2024_08_22.s.sol:20]: Same mistake as the last script. Lets change this to 100 and add some sanity checks that can catch this at the end. (ensuring the total amount of COVE tokens == differences from beginning and the end)

Comment (lilPlumberBoy) [reply script/ops/Ops_2024_08_22.s.sol:20]: ahh same thing as my last multisig pr 😢

File: script/ops/Ops_2024_08_22.s.sol [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        rewardAmounts[2] = 100_000 ether;
+        rewardAmounts[3] = 350_000 ether;
+        rewardAmounts[4] = 50_000 ether;
+        rewardAmounts[5] = 550_000 ether;
+        rewardAmounts[6] = 400_000 ether;
+        rewardAmounts[7] = 100_000 ether;
+        rewardAmounts[8] = 200_000 ether;
+        rewardAmounts[9] = 200_000 ether;
+        rewardAmounts[10] = 200_000 ether;
+        rewardAmounts[11] = 200_000 ether;
```
Comment (penandlim) [inline script/ops/Ops_2024_08_22.s.sol:54]: Let's add a variable coveYFIRewardAmount so we can reuse 850_000

File: script/ops/Ops_2024_08_22.s.sol [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        );
+        addToBatch(
+            coveYfiRewardsGaugeRewardForwarder, 0, abi.encodeCall(RewardForwarder.forwardRewardToken, (coveToken))
+        );
+        require(
+            IERC20(coveToken).balanceOf(coveYfiRewardsGauge) == balanceBefore + uint256(850_000 ether) / 4,
+            "coveYfiRewardsGauge forwardRewardToken failed"
+        );
+
+        uint256 totalRewardsDistributed = 0;
```
Comment (penandlim) [inline script/ops/Ops_2024_08_22.s.sol:69]: This should include rewards to cove YFI

File: script/ops/Ops_2024_08_22.s.sol (hunk 2 above)
Comment (penandlim) [reply script/ops/Ops_2024_08_22.s.sol:54]: Check sum of coveYFIRewardAmount and rewardAmounts == total reward to distribute

File: script/ops/Ops_2024_08_22.s.sol [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        rewardAmounts[7] = 100_000 ether;
+        rewardAmounts[8] = 200_000 ether;
+        rewardAmounts[9] = 200_000 ether;
+        rewardAmounts[10] = 200_000 ether;
+        rewardAmounts[11] = 200_000 ether;
+        uint256 balanceBefore = IERC20(coveToken).balanceOf(coveYfiRewardsGauge);
+        addToBatch(
+            coveToken,
+            0,
+            abi.encodeCall(IERC20.transfer, (coveYfiRewardsGaugeRewardForwarder, uint256(850_000 ether) / 4))
```
Comment (penandlim) [inline script/ops/Ops_2024_08_22.s.sol:59]: Replace references to 850_000 with the new variable

File: script/ops/Ops_2024_08_22.s.sol (hunk 2 above)
Comment (lilPlumberBoy) [reply script/ops/Ops_2024_08_22.s.sol:54]: agreed updated

File: script/ops/Ops_2024_08_22.s.sol (hunk 3 above)
Comment (lilPlumberBoy) [reply script/ops/Ops_2024_08_22.s.sol]: agreed updated

**Review by alphastorm — APPROVED** [review]

**Pull request merged**